support/magicleap/Servo2D/code/src/Servo2D.cpp

```diff
@@ -351,6 +374,33 @@ void Servo2D::urlBarEventListener() {
   navigate_servo(servo_, url_bar_->getText().c_str());
 }
 
+void Servo2D::keyboardVisible(bool visible) {
+  lumin::ui::Keyboard* keys = lumin::ui::Keyboard::Get();
+  if (visible) {
+    lumin::ui::KeyboardProperties properties;
+    properties.keyboardZPosition = lumin::ui::KeyboardProperties::KeyboardZPosition::kVolumeCursorPlane;
+    properties.width = KEYBOARD_W;
+    keys->show(
+      prism_,
+      DEFAULT_LOCALE,
+      properties,
+      std::bind(&Servo2D::keyboardEventListener, this, std::placeholders::_1)
+    );
+  } else {
+    keys->hide();
+  }
+}
+
+bool Servo2D::keyboardEventListener(const lumin::ui::KeyboardEvent::EventData& event) {
+  if (event.getEventType() != lumin::ui::KeyboardEvent::EventType::KEY_PRESSED) {
```

Comment:
Why not pass along KEY_DOWN and KEY_UP like we do in the glutin port, rather than simulating them in libmlservo?

Reply:
Er, because there aren't key down and up, only pressed at https://creator.magicleap.com/learn/reference/public/v0.16.0/LuminRT/namespacelumin_1_1____lr1__0_1_1ui_1_1_keyboard_event.html

Reply:
Checked with magicleap, and there's no way to access the down/up events. https://forum.magicleap.com/hc/en-us/community/posts/360035146792-Keyboard-key-up-and-key-down-events-